client/src/app/components/projects/components/progress-bar/progress-bar.component.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, Input, OnInit } from '@angular/core';
+
+import { environment } from 'src/environments/environment';
+
+const baseUrl = environment.baseUrl;
+
+@Component({
```

Comment:
I guess it will be better to change this component. I guess the component should just get some input values. Because now it can't be reused because contains a lot of logic inside and works directly with data